impeller/entity/geometry.cc

```diff
@@ -16,6 +16,46 @@
 
 namespace impeller {
 
+/// Given a convex polyline, create a triangle fan structure.
+std::pair<std::vector<Point>, std::vector<uint16_t>> TessellateConvex(
+    Path::Polyline polyline) {
+  std::vector<Point> output;
+  output.reserve(polyline.points.size() * 3);
+
+  for (auto j = 0u; j < polyline.contours.size(); j++) {
+    Scalar center_x = 0;
+    Scalar center_y = 0;
+    auto [start, end] = polyline.GetContourPointBounds(j);
+    for (auto i = start; i < end; i++) {
+      const auto& point = polyline.points[i];
+      center_x += point.x;
+      center_y += point.y;
+    }
+    center_x /= end - start;
+    center_y /= end - start;
+
+    for (auto i = start + 1; i < end; i++) {
+      const auto& point_a = polyline.points[i - 1];
+      const auto& point_b = polyline.points[i];
+      output.emplace_back(point_a);
+      output.emplace_back(point_b);
+      output.emplace_back(Point(center_x, center_y));
```

Comment:
Why not define this as a fan, and use `kTriangeFan` below?

Reply:
Triangle fan is not universally supported, and isn't included in the set of PrimitiveTypes